typing/typedtree.ml

```diff
@@ -69,7 +69,7 @@ and expression =
 and exp_extra =
   | Texp_constraint of core_type
   | Texp_coerce of core_type option * core_type
-  | Texp_open of override_flag * Path.t * Longident.t loc * Env.t
+  | Texp_open of override_flag * Longident.t loc * Env.t
```

Comment:
Also, if the Tpat_open, Texp_open, Tcl_open constructors change, perhaps it would be a good opportunity to have them refer to open_description for uniformity (adding the Env.t to open_description).

Reply:
I am not sure about supporting local opens. @yallop and I agree that such extension seems not very useful. However, one paper reviewer, you, and ubsan (in this PR) believe it is nice to have.

Let me see if it is possible to resolve grammar ambiguity first.

Reply:
I have managed to support local open and refactored all of those variants to use `open_description`. I only extended `Pexp_open` this time but not `open`s in class.